Assets/GothicVR/Scripts/Npc/Routine.cs

```diff
@@ -7,44 +7,65 @@
 using GVR.Phoenix.Util;
 using GVR.Util;
 using GVR.World;
+using System;
+using System.Data;
 
 namespace GVR.Npc
 {
     public class Routine : MonoBehaviour
     {
-        private static readonly int SPEED = 10;
-        private GameTime gameTime;
+        private const float SPEED = 1f;
+        private GameTime gameTime = new();
+        PxCs.Data.WayNet.PxWayPointData waypoint;
+        RoutineData routine;
+
         public List<RoutineData> routines = new();
+        public Dictionary<string, RoutineData> waypoints = new();
 
-        void Start()
+        private void OnEnable()
         {
-            gameTime = SingletonBehaviour<GameTime>.GetOrCreate();
+            gameTime.minuteChangeCallback.AddListener(lookUpRoutine);
         }
 
-        void Update()
+        private void Start()
         {
-            if (!SingletonBehaviour<DebugSettings>.GetOrCreate().EnableNpcRoutines)
-                return;
-
-            var routine = GetCurrentRoutine();
+            //Initialize first waypoint
+            DateTime time = new(1, 1, 1, 15, 0, 0);
```

Comment:
This datetime seems like a const. You could think of putting this as a const variable on top of the class.

Reply:
It is just needed to intialize the startposition. so it gets deleted when Start() is finished anyway.

Reply:
I tried. DateTime can't be made const.

Reply:
Yes. It's about "magic numbers": https://en.wikipedia.org/wiki/Magic_number_(programming)

The idea is: If there is a number which has a special meaning, give it a useful name.
To give a name, you need to define a property.

I think of e.g.
```private const DateTime _MinimumDateTime_ = ...```
or
```private const DateTime _StartDateTimeBecause... = ..._```